### PR TITLE
Resolve DXGI_MODE_DESC on any adapter output that holds our window.

### PR DESCRIPTION
**Pull request by bartwe**
Related to issue #140

uses the output that contains our whnd, or the primary monitor as the source of the information used to create the swapchain.

Also resolves an issue with mixing IDXGIAdapter1 and IDXGIAdapter

### CODE REVIEW

File: src/FNA3D_Driver_D3D11.c [hunk 1]
```diff
@@ -5379,18 +5379,55 @@ static void D3D11_PLATFORM_GetDefaultAdapter(
...
 ) {
-	IDXGIOutput *output;
 	IDXGIFactory1* pParent;
 	DXGI_SWAP_CHAIN_DESC swapchainDesc;
 	DXGI_MODE_DESC swapchainBufferDesc;
 	SDL_SysWMinfo info;
 	HWND dxgiHandle;
 	HRESULT res;
 
+
```
Comment (flibitijibibo) [inline src/FNA3D_Driver_D3D11.c:5430]:
```suggestion
```

File: src/FNA3D_Driver_D3D11.c [hunk 2]
```diff
@@ -5404,17 +5441,7 @@ static void D3D11_PLATFORM_CreateSwapChain(
...
-		0,
-		&output
-	);
-	IDXGIOutput_FindClosestMatchingMode(
-		output,
-		&swapchainBufferDesc,
-		&swapchainDesc.BufferDesc,
-		(IUnknown*) renderer->device
-	);
+	ResolveSwapChainModeDescription((IUnknown*) renderer->device, (IDXGIAdapter*) renderer->adapter, (IDXGIFactory1*) renderer->factory, dxgiHandle, &swapchainBufferDesc, &swapchainDesc.BufferDesc);
```
Comment (flibitijibibo) [inline src/FNA3D_Driver_D3D11.c:5444]:
```suggestion
	ResolveSwapChainModeDescription(
		(IUnknown*) renderer->device,
		(IDXGIAdapter*) renderer->adapter,
		(IDXGIFactory1*) renderer->factory,
		dxgiHandle,
		&swapchainBufferDesc,
		&swapchainDesc.BufferDesc
	);
```

**Review by flibitijibibo — APPROVED** [review]
Just whitespace, I'll take care of those...

**Comment by bartwe** [comment]
I am wondering just slightly if we do not want to default to the original behavior, as picking the monitor that starts the application is about as arbitrary a source of settings as is the first monitor on the adapter..

**Comment by flibitijibibo** [comment]
Hypothetically we could pull the window's active monitor via SDL_syswm or something, but this does seem more precise and in most cases will end up on the same display anyhow.

In any case, this is now in fnalibs.

**Pull request merged**